x-pack/plugins/encrypted_saved_objects/server/create_migration.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  SavedObjectUnsanitizedDoc,
+  SavedObjectMigrationFn,
+  SavedObjectMigrationContext,
+} from 'kibana/server';
+import { pick } from 'lodash';
+import { EncryptedSavedObjectAttributesDefinition } from './crypto/encrypted_saved_object_type_definition';
+import { EncryptedSavedObjectTypeRegistration, SavedObjectDescriptor } from './crypto';
+import { EncryptedSavedObjectsMigrationService } from './crypto/encrypted_saved_objects_migration_service';
+
+type SavedObjectOptionalMigrationFn<InputAttributes, MigratedAttributes> = (
+  doc: SavedObjectUnsanitizedDoc<InputAttributes> | SavedObjectUnsanitizedDoc<InputAttributes>,
+  context: SavedObjectMigrationContext
+) => SavedObjectUnsanitizedDoc<MigratedAttributes>;
+
+type IsMigrationNeededPredicate<InputAttributes, MigratedAttributes> = (
+  encryptedDoc:
+    | SavedObjectUnsanitizedDoc<InputAttributes>
+    | SavedObjectUnsanitizedDoc<MigratedAttributes>
+) => encryptedDoc is SavedObjectUnsanitizedDoc<InputAttributes>;
+
+export type CreateESOMigrationFn = <
+  InputAttributes = unknown,
+  MigratedAttributes = InputAttributes
+>(
+  isMigrationNeededPredicate: IsMigrationNeededPredicate<InputAttributes, MigratedAttributes>,
+  migration: SavedObjectMigrationFn<InputAttributes, MigratedAttributes>,
+  inputType: EncryptedSavedObjectTypeRegistration,
+  migratedType?: EncryptedSavedObjectTypeRegistration
+) => SavedObjectOptionalMigrationFn<InputAttributes, MigratedAttributes>;
+
+export const getCreateMigration = (
+  migrationService: EncryptedSavedObjectsMigrationService
+): CreateESOMigrationFn => (
+  isMigrationNeededPredicate,
+  migration,
+  inputType,
+  migratedType = inputType
+) => {
+  if (inputType.type !== migratedType.type) {
+    throw new Error(
+      `An Invalid Encrypted Saved Objects migration is trying to migrate across types ("${inputType.type}" => "${migratedType.type}"), which isn't permitted`
+    );
+  }
+  const inputTypeDefinition = new EncryptedSavedObjectAttributesDefinition(inputType);
+  const migratedTypeDefinition = new EncryptedSavedObjectAttributesDefinition(migratedType);
+  return (encryptedDoc, context) => {
+    if (isMigrationNeededPredicate(encryptedDoc)) {
+      const descriptor = pick<SavedObjectDescriptor, SavedObjectUnsanitizedDoc>(
```

Comment:
__nit:__ let's just manually pick these three properties, we're trying to get rid of `lodash` where it's possible.

Reply:
np 👍